youtube_dl/extractor/facebook.py

```diff
@@ -312,6 +324,22 @@ def extract_video_data(instances):
         if server_js_data:
             video_data = extract_video_data(server_js_data.get('instances', []))
 
+        if not video_data:
```

Comment:
All this branch should go after unavailability checks and before `Cannot parse data`.